solana/restaking/programs/restaking/src/lib.rs

```diff
@@ -68,13 +73,76 @@ pub mod restaking {
             ctx.accounts.token_mint.key()
         );
 
-        staking_params
+        let token_index = staking_params
             .whitelisted_tokens
             .iter()
-            .find(|&&token_mint| token_mint == ctx.accounts.token_mint.key())
+            .enumerate()
+            .find_map(|(index, &token_mint)| {
+                if token_mint == ctx.accounts.token_mint.key() {
+                    return Some(index);
+                }
+                None
+            })
```

Comment:
Sounds like you’re looking for [`position`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.position).

Reply:
oh yes right.